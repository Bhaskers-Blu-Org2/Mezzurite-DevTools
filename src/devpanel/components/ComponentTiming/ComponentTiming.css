@media only screen {
  .component-timing--card {
    align-items: center;
    background-color: var(--color-gray-darker);
    border-radius: 4px;
    color: var(--color-white);
    display: flex;
    justify-content: space-between;
    margin-bottom: 8px;
    text-align: center;
    width: 100%;
  }

  .component-timing--detail {
    display: flex;
    margin: 12px;
  }

  .component-timing--name {
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    margin: 12px;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .component-timing--statistic {
    font-weight: 700;
  }
}

@media only screen and (min-width: 540px) {
  .component-timing--card {
    margin-bottom: 8px;
    margin-right: 2px;
  }

  .component-timing--statistic {
    margin-left: 4px;
  }
}
