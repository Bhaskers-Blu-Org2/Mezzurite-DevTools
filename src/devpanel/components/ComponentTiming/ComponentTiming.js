import React from 'react';
import { number, string } from 'prop-types';

import './ComponentTiming.css';

const ComponentTiming = (props) => (
  props != null && (props.name != null || props.loadTime != null) && <li className='component-timing--card'>
    {props.name != null && <h3 className='component-timing--name'>{props.name}</h3>}
    {props.loadTime != null &&
      <p className='component-timing--detail'>
        <span className='mobile-hidden'>Load time:</span>
        <span className='component-timing--statistic'>{props.loadTime}</span>
        ms
      </p>}
  </li>
);

ComponentTiming.propTypes = {
  loadTime: number,
  name: string
};

export default ComponentTiming;
