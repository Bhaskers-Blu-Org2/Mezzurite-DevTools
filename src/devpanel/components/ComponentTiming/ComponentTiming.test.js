import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ComponentTiming from './ComponentTiming';

describe('ComponentTiming.js', () => {
  const renderer = new ShallowRenderer();

  it('should not render when the props are null', () => {
    const tree = renderer
      .render(<ComponentTiming />);
    expect(tree).toBeFalsy();
  });

  it('should not render when the name and loadTime are null', () => {
    const tree = renderer
      .render(<ComponentTiming loadTime={null} name={null} />);
    expect(tree).toBeFalsy();
  });

  it('should render the componentTiming with only a name', () => {
    const tree = renderer
      .render(<ComponentTiming name='ComponentName' />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the componentTiming with only a loadTime', () => {
    const tree = renderer
      .render(<ComponentTiming loadTime={14.2} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the componentTiming', () => {
    const tree = renderer
      .render(<ComponentTiming loadTime={14.2} name='ComponentName' />);
    expect(tree).toMatchSnapshot();
  });
});
