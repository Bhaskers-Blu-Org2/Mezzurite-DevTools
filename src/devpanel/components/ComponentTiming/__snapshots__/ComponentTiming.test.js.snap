// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTiming.js should render the componentTiming 1`] = `
<li
  className="component-timing--card"
>
  <h4
    className="component-timing--name"
  >
    ComponentName
  </h4>
  <p
    className="component-timing--detail"
  >
    <span
      className="mobile-hidden component-timing--label"
    >
      Load time:
    </span>
    14.2
    ms
  </p>
</li>
`;

exports[`ComponentTiming.js should render the componentTiming with only a loadTime 1`] = `
<li
  className="component-timing--card"
>
  <p
    className="component-timing--detail"
  >
    <span
      className="mobile-hidden component-timing--label"
    >
      Load time:
    </span>
    14.2
    ms
  </p>
</li>
`;

exports[`ComponentTiming.js should render the componentTiming with only a name 1`] = `
<li
  className="component-timing--card"
>
  <h4
    className="component-timing--name"
  >
    ComponentName
  </h4>
</li>
`;
